graphql_compiler/compiler/helpers.py

```diff
@@ -237,6 +237,45 @@ def get_location_name(self):
         """Return a tuple of a unique name of the location, and the current field name (or None)."""
         raise NotImplementedError()
 
+    @abstractmethod
+    def _compare(self, other):
+        """Return True if the other object is smaller than self in the total ordering."""
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        """Return True if the other object is smaller than self in the total ordering."""
+        if isinstance(self, Location) and isinstance(other, Location):
+            return self == other
```

Comment:
I think you can leave `__eq__` as an `abstractmethod`. Otherwise, I think that this line will cause infinite recursion.